model-optimizer/install_prerequisites/install_prerequisites.sh

```diff
@@ -83,18 +83,130 @@ elif [[ "$OSTYPE" == "darwin"* ]]; then
+    else
+      mo_is_custom=false
+      echo "[ INFO ] Release ModelOptimizer version $mo_release_version detected"
+    fi
+
+    if ! check_ie "$python_executable"; then
+        # check pip installed openvino package
+        if check_ov_package "$python_executable" "$requires_sudo"; then
+            if $mo_is_custom; then
+                print_warning "Existing OpenVINO pip version is incompatible with ModelOptimizer."
+                print_warning "For not release ModelOptimizer version please build InferenceEngine python from source (preferable) or install latest OpenVINO version using pip install openvino (may be incompatible)."
+            else
+                print_warning "Existing OpenVINO pip version is incompatible with ModelOptimizer."
+                print_warning "For release ModelOptimizer version which is $mo_release_version please install OpenVNO using pip install openvino==$mo_release_version or build InferenceEngine python from source."
+            fi
+            return 0
+        fi
+
+        print_warning "No available InferenceEngine python API was found. Trying to install OpenVINO using pip."
+
+        if $mo_is_custom; then
+            print_warning "Custom ModelOptimizer version detected."
+            print_warning "The desired version of InferenceEngine can be installed only for release ModelOptimizer version."
+            print_warning "The latest OpenVINO version will be installed (may be incompatible with current ModelOptimizer version)."
+        elif install_ov "$python_executable" "$requires_sudo" "$mo_release_version"; then
+            if check_ie "$python_executable"; then
+                return 0
+            fi
+
+            print_warning "Installed OpenVINO version $mo_release_version doesn't work as expected...Uninstalling..."
+            print_warning "Please consider to build InferenceEngine python from source"
+
+            uninstall_ov "$python_executable" "$requires_sudo"
+            return 0
+        else
+            print_warning "Can not find OpenVINO version $mo_release_version in pip."
+            print_warning "The latest OpenVINO version will be installed (may be incompatible with current ModelOptimizer version)"
+        fi
+
+        #install the latest OpenVINO pip version
+        if install_latest_ov "$python_executable" "$requires_sudo"; then
+            if check_ie "$python_executable"; then
+                return 0
+            else
+                print_warning "Installed latest OpenVINO version doesn't work as expected...Uninstalling..."
+                uninstall_ov "$python_executable" "$requires_sudo"
+                print_warning "Please consider to build InferenceEngine python from source."
+                return 0
+            fi
+        else
+            print_warning "No OpenVINO version is available in pip for installation."
+            print_warning "Please consider to build InferenceEngine python from source."
+            return 0
```

Comment:
Is it an error case? if yes, can we return non 0 return code?

Reply:
We do not check return value of `find_ie_version` function. So return value must be always 0. Also this is not an error. This is a warning.